Patch-Source: https://gitlab.winehq.org/mono/mono/-/merge_requests/102
---
From 3e2d91111d58376b84ac8980a3d9df9a2265348e Mon Sep 17 00:00:00 2001
From: fossdd <fossdd@pwned.life>
Date: Fri, 25 Apr 2025 16:34:58 +0200
Subject: [PATCH] mono-context: Remove use of __GLIBC_PREREQ

__GLIBC_PREREQ is obviously only known in glibc and their use on other
libcs fails like musl libc fails with:

	../../mono/utils/mono-context.h:901:19: error: missing binary operator before token "("
	  901 | #if __GLIBC_PREREQ(2, 26)
	      |                   ^
---
 mono/utils/mono-context.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mono/utils/mono-context.h b/mono/utils/mono-context.h
index 8e423acb398..c4d64fabfb9 100644
--- a/mono/utils/mono-context.h
+++ b/mono/utils/mono-context.h
@@ -898,7 +898,7 @@ typedef struct {
 
 #include <sys/ucontext.h>
 
-#if __GLIBC_PREREQ(2, 26)
+#if __GLIBC__ > 2 || __GLIBC__ == 2 && __GLIBC_MINOR__ >= 26
 typedef ucontext_t MonoContext;
 #else
 typedef struct ucontext MonoContext;
-- 
GitLab

